docs/source/cpp/io.rst

```diff
@@ -73,6 +73,9 @@ The :class:`filesystem interface <FileSystem>` allows abstracted access over
 various data storage backends such as the local filesystem or a S3 bucket.
 It provides input and output streams as well as directory operations.
 
+.. seealso::
+    :ref:`FileSystems API reference <cpp-api-filesystems>`.
```

Comment:
`FileSystem` is the C++ class, but otherwise I think we would just write "Filesystems" or "filesystems".